### PR TITLE
Admin clients/drivers list

### PR DESCRIPTION
**Pull request by ArtyomKosakyan**
Added Clients/Drivers list view functionality for Admin. 
Added Admin menu-bar to the Admin profile page. 
Solved bugs on rails API.

### CODE REVIEW

File: src/components/Admin/ClientField.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  let fields;
+  switch (client) {
+    case 'passengers':
+      try {
+        fields = usersList.users.map(({ passenger }) => {
+          const {
+            // eslint-disable-next-line camelcase
+            id, phone_number, first_name, last_name,
+          } = passenger;
+          return (
```
Comment (iwazowsky) [inline src/components/Admin/ClientField.js:16]: The return statements inside switch cases may be turned into separate components

File: src/components/Client/ClientSettings.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        username: fields.name,
+        password: fields.newPassword,
+      })
+        .then((response) => response)
+        .catch((error) => error);
+      history.push(appRoutes.client);
+    }
+  };
+
+  return (
```
Comment (iwazowsky) [inline src/components/Client/ClientSettings.js]: Big components make debugging harder, try to avoid them

**Review by iwazowsky — APPROVED** [review]
Consider writing more declarative code inside components, didn't notice any critical issues

File: src/components/Client/ClientSettings.js (hunk 2 above)
Comment (ArtyomKosakyan) [reply src/components/Client/ClientSettings.js]: This component's not work yet, It's from first PR Sevak. 
If it does not interfere, we will replace it later.

File: src/components/Admin/ClientField.js (hunk 1 above)
Comment (ArtyomKosakyan) [reply src/components/Admin/ClientField.js:16]: Ok

File: src/redux/actionTypes.js [hunk 3]
```diff
@@ -1,4 +1,7 @@
 export const ACTION_CHANGE_LOGGED_IN = 'ACTION_CHANGE_LOGGED_IN';
 export const ACTION_CHANGE_LOADING = 'ACTION_CHANGE_LOADING';
 export const ACTION_CHANGE_ALERT = 'ACTION_CHANGE_ALERT';
+export const ACTION_ADD_USERS_LIST = 'ACTION_ADD_USERS_LIST';
+export const ACTION_CLEAR_USERS_LIST = 'ACTION_CLEAR_USERS_LIST';
+export const ACTION_ADD_UNVERIFIED_USERS_LIST = 'ACTION_ADD_UNVERIFIED_USERS_LIST';
```
Comment (vnersisyan-vineti) [inline src/redux/actionTypes.js]: remove action work from naming, use simple naming, like `ADD_USERS`, `CLEAR_USERS`

File: src/redux/store.js [hunk 4]
```diff
@@ -43,4 +53,32 @@ const rootReducer = (state = initState, action) => {
       return state;
   }
 };
-export default createStore(rootReducer);
+
+const usersList = (state = initUsersList, action) => {
+  switch (action.type) {
+    case ACTION_ADD_USERS_LIST:
```
Comment (vnersisyan-vineti) [inline src/redux/store.js]: use update action names here

File: src/__test__/API.test.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// import '../utils/API';
+//
+// describe('API testing', () => {
+//   it('user_in ', () => {
+//
```
Comment (vnersisyan-vineti) [inline src/__test__/API.test.js]: remove this file for now

File: src/components/Admin/AdminPage.js [hunk 6]
```diff
@@ -1,20 +1,24 @@
...
+import { useHistory } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import { AdminPageButtons } from '../../utils/configs';
 import { PageButton } from '../Buttons';
 import MenuHelper from '../../helpers/MenuHelper';
+import { mapStateToProps } from '../../redux/actions';
 
-const DriverPage = () => {
+const AdminPage = ({ appState }) => {
```
Comment (vnersisyan-vineti) [inline src/components/Admin/AdminPage.js]: `{ appState: { userId }}`?

File: src/components/Admin/AdminPage.js [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
   const history = useHistory();
-  const location = useLocation();
+  const { userId } = appState;
```
Comment (vnersisyan-vineti) [inline src/components/Admin/AdminPage.js]: this line will be redundant if you destruct props object deeper

File: src/components/Admin/AdminPage.js [hunk 8]
```diff
@@ -1,20 +1,24 @@
...
 
   const onSelect = (event) => {
-    const path = MenuHelper({ event, location });
+    const path = MenuHelper({ event, userId });
```
Comment (vnersisyan-vineti) [inline src/components/Admin/AdminPage.js:14]: separate this lines

File: src/components/Admin/AdminPage.js [hunk 9]
```diff
@@ -1,20 +1,24 @@
...
     history.push(path);
   };
 
   const buttons = AdminPageButtons.map((button) => (
     <PageButton
+      key={button[0]}
```
Comment (vnersisyan-vineti) [inline src/components/Admin/AdminPage.js]: if you change to object declaration you avoid to get first element of button array. **

File: src/components/Admin/AdminProfile.js [hunk 10]
```diff
@@ -18,20 +26,43 @@ const AdminProfile = ({ match }) => {
...
+  }, []);
+
+  const onSelect = (event) => {
+    const path = MenuHelper({ event, userId });
+    history.push(path);
+  };
+
+  const buttons = AdminPageButtons.map((button) => (
+    <PageButton
+      key={button[0]}
```
Comment (vnersisyan-vineti) [inline src/components/Admin/AdminProfile.js]: same comment here

File: src/components/Admin/ClientField.js (hunk 1 above)
Comment (vnersisyan-vineti) [reply src/components/Admin/ClientField.js:16]: @ArtyomKosakyan any progress on this point?

File: src/components/Admin/ClientList.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  const { userId } = appState;
+
+  const onSelect = (event) => {
+    const path = MenuHelper({ event, userId });
+    history.push(path);
+  };
+
+  useEffect(() => {
+    GetUsers({ client, dispatch })
+      .then();
```
Comment (vnersisyan-vineti) [inline src/components/Admin/ClientList.js]: then what?

**Review by vnersisyan-vineti — CHANGES_REQUESTED** [review]
some comments, please address them

File: src/components/Admin/AdminPage.js (hunk 8 above)
Comment (ArtyomKosakyan) [reply src/components/Admin/AdminPage.js:14]: const path = MenuHelper({ event, location });` deleted

File: src/components/Admin/AdminPage.js (hunk 9 above)
Comment (ArtyomKosakyan) [reply src/components/Admin/AdminPage.js]: Changed

File: src/components/Admin/AdminProfile.js (hunk 10 above)
Comment (ArtyomKosakyan) [reply src/components/Admin/AdminProfile.js]: Changed

File: src/components/Admin/ClientList.js (hunk 11 above)
Comment (ArtyomKosakyan) [reply src/components/Admin/ClientList.js]: // eslint-disable-next-line camelcase
            id, phone_number, first_name, last_name,
          } = passenger;
          this solved on #13 PR

**Review by vnersisyan-vineti — APPROVED** [review]
LGTM

**Pull request merged**